packages/azure-http-specs/specs/azure/client-generator-core/client-initialization/client.tsp

```diff
@@ -0,0 +1,185 @@
+import "./main.tsp";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/spector";
+
+using Spector;
+
+namespace _Specs_.Azure.ClientGeneratorCore.ClientInitialization;
+
+model HeaderParamClientOptions {
+  @doc("The name of the client. This parameter is used as a header in all operations.")
+  name: string;
+}
+
+model MultipleParamsClientOptions {
+  @doc("The name of the client. This parameter is used as a header in all operations.")
+  name: string;
+
+  @doc("The region to use for all operations. This parameter is used as a query parameter.")
+  region: string;
+}
+
+model MixedParamsClientOptions {
+  @doc("The name of the client. This parameter is used as a header in all operations.")
+  name: string;
+}
+
+model PathParamClientOptions {
+  @doc("The name of the blob. This parameter is used as a path parameter in all operations.")
+  blobName: string;
+}
+
+model ParamAliasClientOptions {
+  @doc("Blob name for the client.")
+  @global.Azure.ClientGenerator.Core.paramAlias("blob")
+  blobName: string;
+}
+
+// Scenario 1: Header parameter moved to client level
+@scenarioDoc("""
+  Client for testing header parameter moved to client level.
+  
+  Parameters elevated to client level:
+  - name: "test-name-value" (header parameter)
+  
+  Expected client usage:
+  ```ts
+  const client = new HeaderParamClient({
+    name: "test-name-value"
+  });
+  
+  client.withQuery(id: "some-id");  // No need to pass name here
+  client.withBody({ name: "test-name" });  // No need to pass name here
+  ```
+  """)
+@scenario
```

Comment:
can we add scenario doc for each operation? otherwise, it will only be one line in spector dashboard.

Reply:
That's what I was going for, I think it's fair to count being able to do a call with a query and with a body to mean "i can support moving a header parameter onto the client". If you feel strongly, I can change it.

Reply:
idk if spector framework will mark this scenario with green iff all the inner operation pass. if so, i'm ok with it.

Reply:
It will only be green iff all the inner operations pass